app/server/models/token.py

```diff
@@ -43,5 +63,27 @@ def token_amount_to_system(self, token_amount):
         return int(token_amount) * 100 / 10**self.decimals
 
     def system_amount_to_token(self, system_amount):
-        return int(float(system_amount) / 100 * 10**self.decimals)
+        return int(float(system_amount) * 10**self.decimals / 100)
+
+    def add_exchange_contract(self, exchange_contract: ExchangeContract):
+
+        exchange_transfer_account = (server.models.transfer_account.TransferAccount.query
```

Comment:
are exchange transfer accounts like a special thing? they're not associated with any particular user right?

Reply:
Yep. Transfer accounts can be associated with anything that can send/receive tokens. Ie:
- users
- organisations (which are in turn bound to admin users)
- exchange contracts